lib/chef/provider/package/chocolatey.rb

```diff
@@ -136,14 +136,27 @@ def check_resource_semantics!
         #
         # @return [String] full path of choco.exe
         def choco_exe
-          @choco_exe ||=
+          @choco_exe ||= begin
+            path = choco_install_path
+
+            # GetEnvironmentVariable returns "" on failure.
+            if path.to_s.empty?
+              raise Chef::Exceptions::MissingLibrary.new("Could not find Chocolatey. Is it installed?")
```

Comment:
I'm thinking of the kind of case where a user knows just enough to be dangerous and changes some setting that breaks it. If we tell them to run `[System.Environment]::GetEnvironmentVariable('ChocolateyInstall', 'MACHINE')` and it should produce output like `x` then they can repro outside of Chef and figure out what they broke.